tools/RAiDER/aria/prepFromGUNW.py

```diff
@@ -231,6 +232,10 @@ def update_yaml(dct_cfg:dict, dst:str='GUNW.yaml'):
 def main(args):
     """ Read parameters needed for RAiDER from ARIA Standard Products (GUNW) """
 
+    # Check if WEATHER MODEL API credentials hidden file exists, if not create it or raise ERROR
+    credentials.check_api(args.weather_model, args.api_uid, args.api_key,
```

Comment:
@mgovorcin @cmarshak  I wonder if we should add this add the main riader program. The checking of credentials is applicable to all workflows where a model with API is requested?

Reply:
@jlmaurer I also wonder is there a way for us to check without doing a downlaod of data if credentials are correct? aka do the ecmwfapi and cdsa api allow for a credential check?

Reply:
I think raider already checks if the credentials exist and has great documentation on how to setup the hidden credential files. Pretty sure errors are very helpful.

I honestly think the existing approach sufficient/preferred. It aligns well with the existing API libraries too. I think passing credentials each time you want to call raider is probably not preferred choice for a user because when they share commands/code, they are *not* going want to share the credentials (just instructions for their team to update their own hidden credential files).

The main thrust of this is being able to pass environment variables that will be defined when a container is created. I don't know the details of this approach except this is a secure, reliable way to pass credentials for this type of processing.

Reply:
I think this can be included in the main raider workflow. This is why I added the prompt option to remind a user about API credentials in order to use raider.py. The majority of people do not really read all instructions.

This function serves as a double check to see if the hidden file exists, if not it creates it, and it does it **only once** (the first time). Later on, a user does not need to pass the credentials or think about this anymore. In the case of calcDelayGUNW, prompt_flag will be hardcoded to False, as we do not want to get stuck by waiting for the input.

For the downloading error check, maybe we can add one test dataset to download per model, and try to download it. I do not know if this is a good option as this will run every time when we wanna run raider. I think it good approach to report error if files cannot be downloaded and point user to check his credentials.

Reply:
Got it, perhaps we could add a seperate script stand-alone from the raider workflow where we do this.
e.g. script --model 
- the script checks if file exist, if not checks env variables, if nto pings users
- then script makes a dummy call to the API to check if credentials work and report out.
Half of the functionally exist with what you have already.
The dummy check i would like to see if there is a potentially handschake only option rather than fetching data.
 
Inclusion in the main raider code to see if the file exist would be good for that model being requested and error out if needed.

For example our HRES credentials cycle and you will want to catch that early.